### ISSUE TITLE
Conflicting security-declarations

### ISSUE THREAD

**Issue by ida** [plone/plone.app.blob#25]
Moved from https://dev.plone.org/ticket/20289

Relates to https://github.com/plone/plone.app.blob/issues/22 which was fixed by david-batranu, but commentlessly reverted again.

Description
-----------

Since of Plone-4.3.5, rising an instance yields:

    2015-08-10 10:23:51 INFO ZServer HTTP server started at Mon Aug 10 10:23:51 2015
        Hostname: 0.0.0.0
        Port: 8080
    2015-08-10 10:23:53 WARNING SecurityInfo Conflicting security declarations for
    "<AccessControl.SecurityInfo.ClassSecurityInfo object at 0x5ddef10>"
    2015-08-10 10:23:53 WARNING SecurityInfo Conflicting security declarations for
    "<AccessControl.SecurityInfo.ClassSecurityInfo object at 0x5ddef10>"
    2015-08-10 10:23:53 WARNING SecurityInfo Class "BlobWrapper" had conflicting security declarations
    2015-08-10 10:23:53 WARNING SecurityInfo Conflicting security declarations for
    "<AccessControl.SecurityInfo.ClassSecurityInfo object at 0x5dd4690>"
    2015-08-10 10:23:53 WARNING SecurityInfo Conflicting security declarations for
    "<AccessControl.SecurityInfo.ClassSecurityInfo object at 0x5dd4690>"
    2015-08-10 10:23:53 WARNING SecurityInfo Class "BlobField" had conflicting security declarations

Observations
-----------------

- The warning has been introduced with plone.app.blob-1.5.11. in field.py, where the security-declarations were changed to use decorators, now. One can counterproof that, by simply copying field.py of vs. 1.5.10 over to 1.5.11.

- The history of plone.app.blob misses entirely entries for the time-range between the release of 1.5.10 and 1.5.11 _!!!_

- Version 1.5.12 then claims to be a "Rerelease for clarity because of double release of 1.5.11."

- There used to be the same warning-message concerning topics in the Plone-4.1-series, which was silenced – not fixed – by hannosch in: https://github.com/plone/Products.ATContentTypes/commit/1f88839d6bad753a25884d00b4eb4ce31c36dedd

Questions
--------------

- Why is the history of an entire version missing?

- Does double-release mean, there are two eggs existing, with the same version-number?

- How can one get a full list of all releases of an egg, on pypi's website e.g, or otherwise, to see the other half of the doubled release?

- Is silencing the warning considered to be an acceptable resolvement in this case?

- Is there possibly a guide, what needs to be regarded, when changing security-declas to use decorators?

- And of course: What could be a possible resolvement for the warning? I tried to mimicry hannosch's topics-fix in ATContentTypes/field.py, no luck.

Regards, Ida

**Comment by vangheem** [comment]
It was fixed here: https://github.com/plone/plone.app.blob/commit/7cb4aed5d3e01367999e59ccd44d4af80b03a206

**Comment by vangheem** [comment]
The history does look odd but it is correct. It is just hard to follow because some things were reverted and then re-applied when tests were fixed.

**Comment by ida** [comment]
I'm afraid, it wasn't fixed entirely, the warning still shows up when using vs. 1.5.16, reopening again.

The history doesn't look odd, it is *completely missing* for the timeframe between releases, so e.g. one cannot know, who did the changes and it makes VCS pretty senseless.

**Comment by ida** [comment]
I admit the lack of understanding the issue, but warnings about security-issues gives you a nervous feeling. Also I love it, when my prompt just shows four lines after rising the instance: Server started, Host, Port, Ready for requests :-)

**Comment by vangheem** [comment]
I don't see any problems with versioning.

**Comment by ida** [comment]
When there's a change, it should be tracked, or not?
Is it considered to be a "harmless", neglectable warning?

**Comment by vangheem** [comment]
I didn't notice any changes that weren't tracked. Not sure what you're talking about in regard to differences between commits and changelog entries though or what problem you think there is.

**Comment by ida** [comment]
Well if you read my initial lines, it's about heavy modifications in field.py, but there's no commit-msg. Don't get me wrong, it's about figuring out what happened, to be able to avoid it in the future. Maybe a git-workflow thing...

And it would be _really good to know_, if the message means something "serious" to consider, or is neglectable.

Really sorry, not to be able to fix it myself and not bothering you any further with it. I know, there's a lot of other things to do :-)

**Comment by ida** [comment]
P.S.: I ment figuring out what happened concerning the warning, not the history. The first occurs repeatedly when switching to decorators, the latter doesn't, I hope.

**Comment by ida** [comment]
@vangheem 
Because you're not sure what I'm talking about in regard to differences between commits and changelog entries, I'll explain it alittle further for you:

These are the release dates:
1.5.10 (2014-04-16)
1.5.11 (2015-04-30)
Which gives a time-frame from the 16th to 30th of April in 2014, where changes had been done and finally released.
The history however, doesn't show *any* commits in between that range, the closest before is of March 13th, titled "fix tests to work with barceloneta theme" and then the next is of 27th June, titled "ported to plone.app.testing"...


Because you're not sure what problem I think there is:

I don't know if there is a problem, I'm repeatedly asking you, whether there is one, or not, with no luck. And I suppose some more users have this question too, reading about security-conflicts, like the initial reporter on dev.plone.org. Personally I couldn't care less, if this is closed or not, so go ahead.

**Comment by vangheem** [comment]
One of those dates is 2014, the other 2015. So over a year passed between releases. I think you are looking for a 2 week window when it's actually 1 year and 2 weeks.

**Comment by ida** [comment]
Oh gee, you're right, I overlooked it, didn't expect such a long break in dev.
Anyway warning persists, still with newest vs. 1.5.16, you should be able to reproduce it.
Sorry to be so buggy about it, but the thing is merely about, if it's safe to communicate, that this warning doesn't mean any harm. (Which I think, after having a look, at related bug-reports, but am not sure.) Cheers, Ida

**Comment by ida** [comment]
@vangheem Ok, I just did everywhere in BlobWrapper(), what you did with the fix for one field, and replaced `@security.private` with `@security.protected(View)` and that worked! If you'd give green light, that'll be a correct thing to do, for everywhere inside of the classes (`BlobField()`, too) , I see, if can can provide a PR these days.

**Comment by vangheem** [comment]
Not everything should be @security.protected(View)

The fix for those declaration conflicts are fixed with a newer version of AccessControl: https://pypi.python.org/pypi/AccessControl

See 3.0.9

Next plone 4.3 will have this version. Funny, I even updated it and forgot about it: https://github.com/plone/buildout.coredev/commit/d76529dcdb427655f716488678b61d15abc5cfca

**Comment by ida** [comment]
Oh, very nice, thanks a bunch Nathan! I see, having so much to do, one cannot know everything by heart, just good to know, where to look it up ;-)
